multi-agent/adapters/outbound/mongo/builtin_resource_descriptor_repository.py

```
@@ -0,0 +1,172 @@
+        elif ownership == ResourceOwnership.CUSTOM:
+            cond = {**identity_cond, "_descriptor": None}
+        else:
+            cond = {
+                "$or": [
+                    identity_cond,
+                    {"_descriptor.visibility": ResourceVisibility.PUBLIC.value},
+                ]
+            }
+        pipeline.append({"$match": cond})
+
+        count_result = list(self._resources_col.aggregate(pipeline + [{"$count": "total"}]))
```

Ruff (0.16.1)

[warning] Consider `[*pipeline, {"$count": "total"}]` instead of concatenation

Replace with `[*pipeline, {"$count": "total"}]`

(RUF005)